finetune/base_models/gpt/featurizer.py

```diff
@@ -174,10 +178,22 @@ def gpt_featurizer(X, encoder, config, train=False, reuse=None):
                                              scope='h%d' % layer, train=train_layer, scale=True)
                 if config.low_memory_mode and train_layer:
                     block_fn = recompute_grad(block_fn, use_entire_scope=True)
-                h = block_fn(h)
+                if layer < config.n_layer - 1:
+                    h = block_fn(h)
+                else:
+                    h_out = block_fn(h)
+
+            # get the attention weights from the last layer
+            if layer == config.n_layer - 1:
+                with tf.variable_scope('h%d_' % layer, reuse=True):
+                    with tf.variable_scope('h%d' % layer, reuse=True):
+                        with tf.variable_scope('attn', reuse=True):
+                            q, k, v = multihead_qkv(h, n_state=shape_list(h)[-1], n_head=config.n_heads, train=train)
```

Comment:
Doesn't this re-do the computation when it's structured this way?  Even though the weights are shared?